### PR TITLE
adding test to context save/restore backing DRAM memory

### PR DESCRIPTION
**Pull request by jleidel**
This test has an example of how to utilize a binary file to context save/restore the contents of the memory.  The test creates a binary file using `dd`, executes an application and saves the file back off.  The file is mapped into the DRAM memory using `mmap`.  We ensure the file changes post-run using SHA1 hashes.

### CODE REVIEW

File: test/backingstore/backingstore.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * ex2.c
```
Comment (leekillough) [inline test/backingstore/backingstore.c:2]: `backingstore.c`

File: test/backingstore/rev-backingstore.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+#
+# Copyright (C) 2017-2023 Tactical Computing Laboratories, LLC
+# All Rights Reserved
+# contact@tactcomplabs.com
+#
+# See LICENSE in the top level directory for licensing details
+#
+# rev-test-cache2.py
```
Comment (leekillough) [inline test/backingstore/rev-backingstore.py:8]: `rev-backingstore.py`

File: test/backingstore/rev-backingstore.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+#
```
Comment (leekillough) [inline test/backingstore/rev-backingstore.py:1]: Does it need a separate `.py` file considering the new, more universal `rev-model-options-config.py`?   It looks complicated and specific for this test, but that would be a question I'd ask for every new test.

File: test/backingstore/backingstore.c (hunk 1 above)
Comment (jleidel) [reply test/backingstore/backingstore.c]: fixed in 8f113a9

File: test/backingstore/rev-backingstore.py (hunk 2 above)
Comment (jleidel) [reply test/backingstore/rev-backingstore.py]: fixed in 8f113a9

File: test/backingstore/rev-backingstore.py (hunk 3 above)
Comment (jleidel) [reply test/backingstore/rev-backingstore.py:1]: for this test, its fairly vital to maintain a nonstandard config file

**Review by leekillough — APPROVED** [review]

**Pull request merged**